neuraxle/pipeline.py

```diff
@@ -472,3 +472,116 @@ def _create_sub_pipelines(self) -> List['MiniBatchSequentialPipeline']:
                 )
 
         return sub_pipelines
+
+
+class Barrier(NonFittableMixin, NonTransformableMixin, BaseStep, ABC):
+    """
+    A Barrier step to be used in a minibatch sequential pipeline. It forces all the
+    data inputs to get to the barrier in a sub pipeline before going through to the next sub-pipeline.
+
+    ```
+    p = MiniBatchSequentialPipeline([
+        SomeStep(),
+        SomeStep(),
+        Barrier(), # must be a concrete Barrier ex: Joiner()
+        SomeStep(),
+        SomeStep(),
+        Barrier(), # must be a concrete Barrier ex: Joiner()
+    ], batch_size=10)
```

Comment:
This could go in a new `/examples/mini_batch_sequential_pipeline.py` file in the examples gallery and link to it.

Reply:
added issue for this will be done in the next pr #126